zlib-rs/src/deflate/slide_hash.rs

```diff
@@ -26,8 +26,10 @@ fn slide_hash_chain(table: &mut [u16], wsize: u16) {
 
 mod rust {
     pub fn slide_hash_chain(table: &mut [u16], wsize: u16) {
-        for m in table.iter_mut() {
-            *m = m.saturating_sub(wsize);
+        for chunk in table.chunks_exact_mut(16) {
```

Comment:
Maybe add a comment explaining why the size of table is always a multiple of 16.